### PR TITLE
Clover viewer

### PR DESCRIPTION
**Pull request by lfarrell**
https://unclibrary.atlassian.net/browse/BXC-4750

### CODE REVIEW

File: web-services-app/src/main/java/edu/unc/lib/boxc/web/services/processing/IiifV3ManifestService.java [hunk 1]
```diff
@@ -60,7 +62,8 @@ public class IiifV3ManifestService {
     public static final String DURATION = "duration";
     public static final String WIDTH = "width";
     public static final String HEIGHT = "height";
-    private static final List<String> FILE_TYPES = Arrays.asList(VIDEO_MP4.toString(), AUDIO_MP4.toString(), AUDIO_MPEG.toString());
+    private static final List<String> FILE_TYPES = Arrays.asList(VIDEO_MP4.toString(), VIDEO_MPEG.toString(),
```
Comment (bbpennel) [inline web-services-app/src/main/java/edu/unc/lib/boxc/web/services/processing/IiifV3ManifestService.java:65]: hmm wonder if there are going to be conflicts with Kara's m4a ticket, probably a good reason to get this merged in

**Review by bbpennel — CHANGES_REQUESTED** [review]

File: web-common/src/main/java/edu/unc/lib/boxc/web/common/utils/DatastreamUtil.java [hunk 2]
```diff
@@ -87,11 +86,13 @@ public static boolean originalFileMimetypeMatches(ContentObjectRecord metadata,
         if (formatTypes == null || formatTypes.isEmpty()) {
             return false;
         }
-        String mimetype = formatTypes.get(0);
-        if (StringUtils.isBlank(mimetype)) {
-            return false;
+
+        for (String format : formatTypes) {
```
[the remark this answers is not in the document]
Comment (lfarrell) [reply web-common/src/main/java/edu/unc/lib/boxc/web/common/utils/DatastreamUtil.java:90]: Doesn't this one do that https://github.com/UNC-Libraries/box-c/blob/2260bde321bdfed7df620eb820b8601646d158f3/web-common/src/test/java/edu/unc/lib/boxc/web/common/utils/DatastreamUtilTest.java#L116. I've added a check for the text file as well.

[the remark this answers is not in the document]
Comment (bbpennel) [reply web-common/src/main/java/edu/unc/lib/boxc/web/common/utils/DatastreamUtil.java:90]: Oh yeah, I looked at that test but missed that there were two mimetypes on the same line. Maybe we could update the name of the method to indicate that it is testing for multiple mimetypes?

File: search-solr/src/main/java/edu/unc/lib/boxc/search/solr/filters/HasValuesFilter.java [hunk 3]
```diff
@@ -0,0 +1,43 @@
...
+        if (fieldValue.size() == 1) {
+            return filter.toString();
+        }
+
+        /* Start looping from the second value in the field, since we've already got the first value */
+        for (String value : fieldValue.subList(1, fieldValue.size())) {
+            filter.append(" OR ").append(getFieldKey().getSolrField()).append(":").append(value);
+        }
+
+        return filter.toString();
```
Comment (bbpennel) [inline search-solr/src/main/java/edu/unc/lib/boxc/search/solr/filters/HasValuesFilter.java:36]: Could probably be simplified with a stream:
```
fieldValue.stream().map(v -> getFieldKey().getSolrField() + ":" + v)
                               .collect(Collectors.joining(" OR "));
```

File: search-solr/src/main/java/edu/unc/lib/boxc/search/solr/filters/HasValuesFilter.java [hunk 4]
```diff
@@ -0,0 +1,43 @@
...
+
+/**
+ * Filter which restricts results to entries which contain populated values for the given key with
+ * the specified field search value
+ *
+ * @author lfarrell
+ */
+public class HasValuesFilter implements QueryFilter {
+    private final SearchFieldKey fieldKey;
+    private final List<String> fieldValue;
```
Comment (bbpennel) [inline search-solr/src/main/java/edu/unc/lib/boxc/search/solr/filters/HasValuesFilter.java:16]: Should rename to "fieldValues"

**Review by bbpennel — APPROVED** [review]

**Pull request merged**